src/reducer.js

```diff
@@ -320,11 +323,38 @@ const reducer = (state = initialState, action) => {
       return onWalletReloading(state);
     case types.SHARED_ADDRESS_UPDATE:
       return onSharedAddressUpdate(state, action);
+    case types.SET_UNLEASH_CLIENT:
+      return onSetUnleashClient(state, action);
+    case types.SET_FEATURE_TOGGLES:
+      return onSetFeatureToggles(state, action);
+    case types.FEATURE_TOGGLE_INITIALIZED:
+      return onFeatureToggleInitialized(state);
     default:
       return state;
   }
 };
 
+const onFeatureToggleInitialized = (state) => ({
+  ...state,
+  featureTogglesInitialized: true,
+});
+
+/**
+ * @param {Object} action.payload The key->value object with feature toggles
+ */
+const onSetFeatureToggles = (state, { payload }) => ({
+  ...state,
+  featureToggles: payload,
```

Comment:
It would be great, but `unleash-proxy-client` does not provide a "diff" of what toggles changed, it just emits an `update` event and updates its internal state, which is what we fetch (`unleashClient.toggles`) and call `setFeatureToggles` with

You can see this logic on the `handleToggleUpdate` saga in `src/sagas/featureToggle.js`

Reply:
Just a clarification: Even though it does not provide a "diff" of what changed, it uses `ETAG` on the request to only emit the `UPDATE` event if some flag changed state